src/compiler/transformers/classFields.ts

```diff
@@ -664,23 +678,57 @@ namespace ts {
             //      this.x = 1;
             //  }
             //
-            if (constructor?.body) {
-                let afterParameterProperties = findIndex(constructor.body.statements, s => !isParameterPropertyDeclaration(getOriginalNode(s), constructor), indexOfFirstStatement);
-                if (afterParameterProperties === -1) {
-                    afterParameterProperties = constructor.body.statements.length;
+            // If we do useDefineForClassFields, they'll be converted elsewhere.
+            // We instead *remove* them from the transformed output at this stage.
+            let parameterPropertyDeclarationCount = 0;
+            if (constructor && constructor.body) {
+                if (useDefineForClassFields) {
+                    statements = statements.filter(statement => !isParameterPropertyDeclaration(getOriginalNode(statement), constructor));
                 }
-                if (afterParameterProperties > indexOfFirstStatement) {
-                    if (!useDefineForClassFields) {
-                        addRange(statements, visitNodes(constructor.body.statements, visitor, isStatement, indexOfFirstStatement, afterParameterProperties - indexOfFirstStatement));
+                else {
+                    for (const statement of constructor.body.statements) {
+                        if (isParameterPropertyDeclaration(getOriginalNode(statement), constructor)) {
+                            parameterPropertyDeclarationCount++;
+                        }
+                    }
+                    if (parameterPropertyDeclarationCount > 0) {
+                        const parameterProperties = visitNodes(constructor.body.statements, visitor, isStatement, indexOfFirstStatementAfterSuper, parameterPropertyDeclarationCount);
+
+                        // If there was a super() call found, add parameter properties immediately after it
+                        if (foundSuperStatement) {
+                            addRange(statements, parameterProperties);
+                        }
+                        // If a synthetic super() call was added, add them just after it
+                        else if (needsSyntheticConstructor) {
+                            statements = addRange(
+                                [statements[0]],
+                                [
+                                    ...parameterProperties,
+                                    ...statements.slice(1),
+                                ]
+                            );
+                        }
+                        // Since there wasn't a super() call, add them to the top of the constructor
+                        else {
+                            statements = addRange(

```

Comment:
Why are you using `addRange` with an empty array here? Why not just do `[...parameterProperties, ...statements]`?